backend/services/implementations/classService.ts

```diff
@@ -198,6 +199,10 @@ class ClassService implements IClassService {
           `Class with id ${id} not found or not currently active`,
         );
       }
+
+      // Make a best effort to delete upcoming test sessions and end active test sessions
+      await this.deleteUpcomingTestSessions(id, date);
+      await this.endActiveTestSessions(id, date);
```

Comment:
Maybe make a note that the `await`ed methods here catch any exceptions, which is why this is best-effort?